mne/preprocessing/annotate_nan.py

```diff
@@ -0,0 +1,41 @@
+# Author: David Julien <david.julien@ifsttar.fr>
+#
+# License: BSD (3-clause)
+
+from ..annotations import (Annotations)
+
+
+def annotate_nan(raw):
+    """Detect segments with NaN and annotate them.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to find segments with NaN values.
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with NaN segments annotated as NAN
+    """
+    annot = raw.annotations
+
+    onsets = []
+    durations = []
+    desc = []
+    sampling_duration = 1 / float(raw.info['sfreq'])
+    data, times = raw.get_data(return_times=True)
+    for row, channel_values in enumerate(data):
+        for time, value in zip(times, channel_values):
+            if str(value) == 'nan':
+                print("nan @ %d" % time)
+                onsets.append(time)
+                durations.append(sampling_duration)
+                desc.append('nan')
+
+    if annot is None:
+        annot = Annotations(onsets, durations, desc)
+    else:
+        annot.append(onsets, durations, desc)
+
+    return annot
```

Comment:
Should we annotate `raw` inplace, instead of returning an **Annotations** instance containing the existing annotations and the new ones ?

Reply:
I think in some other issue recently we decided to return a new annotations object rather than modify in place